### ISSUE TITLE
v0.56a: OpenCV Error: Bad argument (Input array is not a valid matrix) in cvPointSeqFromMat, file c:\builds\2_4_PackSlave-win32-vc12-shared\opencv\modules\imgproc\src\utils.cpp, line 54

### ISSUE THREAD

**Issue by touchhope** [Ractiv/touch_plus_source_code#8]
Hello Lai Schwe,

The hungry of hope frog has catch a bug for you on v0.56a : "OpenCV Error: Bad argument (Input array is not a valid matrix) in cvPointSeqFromMat, file c:\builds\2_4_PackSlave-win32-vc12-shared\opencv\modules\imgproc\src\utils.cpp, line 54".
See http://gharbi.me/ractiv/ for details on error context.

Thank you for publishing your code and congratulations for hard work.

If Darren is Ractiv's webmaster, please tell him to take the time to publish a little update on the blog.
For windows installer, it's unlogical that it refuse to install under win 7 x64.
Please clearly define dependancies for msvcr110.dll and msvcp110.dll on win 10 x32.
To fix : loss of device detection just before keyboard step that get you to replug to go on.
4 instances of win_cursor_plus.exe launched by deamon_plus.exe that make win_cursor_plus_fallback.exe crashing.
To fix : updater that download patch but fail to install update
To add : a new menu on deamon to allow user to ask for a new initialization (instead of manually deleting folders).

Glad to see this marvellous project is still active (for Ractiv it's normal, no ?)

Best regards,

The frog (known as touchhope on Wassim's website)

**Comment by touchhope** [comment]
v0.56b : This time updating software did work automatically. But still the same issue of win_cursor_plus_fallback.exe crashing with 4 instances of win_cursor_plus.exe launched by deamon_plus.exe. This version is worse because i even can't get the two little windows named "image_foreground0" and "image_foreground1" and don't get to keyboard step.

**Comment by sheley1998** [comment]
Getting on it.

**Comment by bogdi1988** [comment]
I'm seeing some bugs on getting the new code compiled in VS2013. Reference
to the following file missing:
Error 2 error C1083: Cannot open include file: 'opencv2/opencv.hpp': No
such file or directory
(..\..\track_plus_core\track_plus\camera_initializer_new.cpp)
c:\users\alexa\documents\github\touch_plus_source_code\track_plus_core\track_plus\camera_initializer_new.h
21 1 track_plus


Alexandru Bauer

**Comment by touchhope** [comment]
Thank you !

**Comment by bogdi1988** [comment]
Also this error: 
Error	1	error C1083: Cannot open include file: 'SFML/Network.hpp': No such file or directory (..\..\track_plus_core\daemon_plus\udp.cpp)	c:\users\alexa\documents\github\touch_plus_source_code\track_plus_core\daemon_plus\udp.h	21	1	daemon_plus

**Comment by bogdi1988** [comment]
I take it back... noob error. Added the environment variables properly and got rid of those. However I get some syntax and some missing type specifier errors at build.

**Comment by bogdi1988** [comment]
I'm getting 4 instances of menu_plus.exe

**Comment by sheley1998** [comment]
The menu is running in Node WebKit, the runtime automatically creates 4 instances. Please check back in 24 hours, I've done a lot of work lately that lead the current commit to become a lot less reliable. Working to get it smoothed out.

**Comment by bogdi1988** [comment]
Thank you! Looking forward to compiling and testing out the next build (now that VS is set up right and compiles all code ok)

**Comment by touchhope** [comment]
v56c: See http://gharbi.me/ractiv/ 
@TF
Just first download "Touch+ Software.exe" from the link provided by Xiangyu Meng 5 days ago :

I'm a buyer from China,here is a link of touch+ software 

http://pan.baidu.com/s/1pJ0zNfl

Then when deamon_plus.exe is running, right click one the little icon on you task bar and choose "Update software" to migrate to version 0.56c.

In this current version 0.56c again for me it seems worse than v0.56a.

In v0.56c, i get to keyboard step but when i push 8 key, it does nothing (oops ! qwerty and azerty automatic detection added ?), no when i switch to azerty again, it detects my 8 key but block on = key !

I don't obtain the two little windows and before keyboard step, i have had a lot of win_cursor_fallback.exe crash again and have had often to replug device !

After pushing 8 key, in the console i have "message sent: menu_plus show calibration next step 0 complete" and then 5 to 5 numbers and nothing (blocked on = key, no matter if azerty or qwerty).

When showing in console "pose estmator initialized, message sent: win_cursor_plus open udp channel", and also first with "menu_plus created, message_received get toggles deamon_plus1, message sent: menu_plus get toggles 1011005" to go on out of the blocking you have to replug device !

Then you go to taskmgr, to kill some of the 4 instances of win_cursor_plus.exe in memory, just to get the control panel to come again. The console runs again and says menu_plus created but nothing appear !

@Lai Schwe:
Wouldn't be much more simple if you could temporaly bring debugging tool to the build automatically sending you errors codes ?

**Comment by sheley1998** [comment]
Hi,
It seems like many folks are trying to run the software in Windows 7. The
cursor code for Windows 7 and 8&above are different because 7 does not
natively support touchscreens and touch-injection. I haven't updated the
Win7 cursor for a while already, and as a result it is extremely bugged up.
I'm downloading Win7 at the moment, and will start working on it once I
have the OS installed in a virtual machine.

Thanks

**Comment by sheley1998** [comment]
Will give you a heads up once I've gotten it smoothed out.

**Comment by touchhope** [comment]
Thanks.
Indeed, i was trying under win 7 x64 (installer refuse to install under this os and i have copy the files from win 8.1 on win 7).
I will try the new build in win 8.1 after lunch at noon.

**Comment by touchhope** [comment]
Is it important to install and run as administrator or not, because we see a little icon on the program icon which suggest administrator privileges needed, no ?

**Comment by touchhope** [comment]
current v[e]sion v0.57a (Do you mean current ve[r]sion in the bubble ?)
Well, i've unstalled completely the software under win 8.1 x64 and then reinstall and of course first update don't work.
You must have the first patch.zip after v.55b to first overwrite with v0.56a to make update work to get v0.57a (and as deamon_plus.exe was killed you have to kill also one of the 4 instances ofmenu_plus.exe because "Exit" menu doesn't work at this step).
A bit tedious as someone said !
Then i'm sorry but again under win 8.1 x64, first a big "An unhandled win32 exception occurrredintrack_plus.exe [3648]".
It says "elevated" talking about the profile running the program.
When you replug device, the console goes on with messages.
How long must you wiggle your fingers ?
Because i was fed up to wiggle (blocked on last console message "exposure is") and then starting to write this on gihub and as i already saw it on win 7 x64, suddenly the two little windows appear and i see my two hands moving inside !
So you're then on keyboard step but this time with qwerty switch, after pushing successfully on 8 and = keys, i keep being blocked on shift key (better than previous but still blocked !).

**Comment by touchhope** [comment]
In order to lauch a new initialization after updating software, you sometimes need to delete userdata folder and ipc too i'm not sure ?

**Comment by touchhope** [comment]
And sometimes, you go and search in the first patch.zip the database folder because it seems it's more recent and fonctional ?

**Comment by touchhope** [comment]
see : http://gharbi.me/ractiv/index.php

**Comment by touchhope** [comment]
lol ! :) super excited !

**Comment by touchhope** [comment]
Could you tell me what's the problem with these two dll files msvcr110.dll and msvcp110.dll that touch+ software doesn't find to run under win 10 x32 ?

**Comment by touchhope** [comment]
When you and Darren will be billionaires with touch+ device, would you invite us all to eat at La Tour d'Argent or Chez Maxime if you prefer in Paris to celebrate that ?

**Comment by touchhope** [comment]
lol ! :)

**Comment by touchhope** [comment]
@Alexandru
 Again i beg you : could you share your msvcr110.dll and msvcp110.dll files on Windows 10 build 10240 for win 10 x32 and now i only got win 10 x64 computers your msvcr120.dll and msvcp120.dll because you know what : touch+ software don't work on public family win 10 x64 ! Must i have to install vs_shell_iso32 runtime to make it work ? 
 @Lai Schwe
 You'll have to find other buyers or backers until you fix your installer !

**Comment by bogdi1988** [comment]
@touchhope - i used this on Windows 10 64bit. I did NOT have to put those DLLs in the system. I just installed the app and it just worked.

**Comment by touchhope** [comment]
@Alexandru & @Lai Schwe
 I confirm, as two dll are missing in the installer, if you want have it running ubder win 10 x64 family, you have to vs_isoshell.exe (Microsoft Visual Studio 2013 runtime isolated).
 Again with v0.57a, tedious slow initialization and blocking on 8 key no matter qwerty or azerty and still no pointing circle !

**Comment by touchhope** [comment]
After a bad night, dreaming of a working touch+, new try this morning with the idea of taking care of caps lock enabling : keyboard step ok but make sure you wait until 95 for each key in the console. I get the 2 little windows but still no pointing circle but again a big crash with error message from vs debugger "An unhandled win32 exception occurred in track_plus.exe".

**Comment by touchhope** [comment]
I don't know why on one of my 2 pc (the ex win 8.1 x64) under win 10 x64 when i click on "Update software" it does nothing but on the other, which was previously under win 7 x64, where i have had to install vs_isoshell.exe to get the 2 dll, asking for update still work.

**Comment by sheley1998** [comment]
Yeah I've started experiencing issues myself after updating my laptop. Still installing latest visual studio, will start debugging in a bit.

**Comment by touchhope** [comment]
Got now the 2 little windows running and no crash for now (few minutes) but after deleting userdata folder and ipc for new initialization, but still no pointing circle !

**Comment by touchhope** [comment]
Deamon_plus.exe console says "step 3 complete, message sent: menu_plus show stage, calibration finished". Still no crash : a kind of stability ! lol :)

**Comment by sheley1998** [comment]
Gimme half a day to get things sorted. BTW what model is your laptop?

**Comment by touchhope** [comment]
ACER ASPIRE R7 WITH EZEL HINGE

**Comment by touchhope** [comment]
Thank you and good afternoon, is that right ? Here, it's the morning in France.

**Comment by touchhope** [comment]
Well restarting after clicking on "Exit", then again blocked after "pose estimator initialized, message sent: win_cursor_plus open udp channel" and this time replug don't make anything on a eventually loss of device detection.

**Comment by touchhope** [comment]
In fact device detection was completely broken and i had to click on Exit, then replug and hearing windows sound that tell it detects something on usb.

**Comment by touchhope** [comment]
How long should we wiggle normally ? Because the two next step of gestures after are so fast that we don't have time to move properly i think.

**Comment by sheley1998** [comment]
Normally, around 2 seconds. Here try the latest build from this link https://www.dropbox.com/sh/41v0qwqysth6dfl/AABNSRVG6rEdHrvDUgd7z6wga?dl=0

**Comment by sheley1998** [comment]
Try terminating all running executables from the old build if you run into any issues

**Comment by touchhope** [comment]
Thank you for that new build.
Still no more working update menu.
Well at exposure is 10 it blocks then after few minutes, show wiggle message, i wiggle for a half minute but nothing and suddenly the 2 next gesture steps and then the keyboard step but very very slow hard to get to 95 for each key then the 2 little windows and still no pointing circle !

**Comment by sheley1998** [comment]
Alight just needed to verify that the proble exists across different builds. Thanks for reporting these issues, I'm getting on it.

**Comment by bogdi1988** [comment]
Tried this build you just posted on dropbox.
App starts... i get the following: track_plus.exe, win_cursor_plus.exe and 4 instances of menu_plus.exe
I gets to the wiggle screen and can't get past it. Camera lights are on, however it doesn't seem to see any input.

**Comment by bogdi1988** [comment]
I take it back. I deleted all the folders, re-copied the ones from the new build. Started the app again, let it download the calibration data and I was able to pass the wiggle part. Took about 30 seconds on the wiggle part until it finally kicked the camera process. Finger pointer movement is choppy and still only works in a really tiny area. Screenshot attached.
Also, the menu for the control panel has no scroll function so I can't really see the whole menu. This is on a 4K monitor - seems the app does not like higher DPI screen. I will try to run the app in smaller DPI
![capture](https://cloud.githubusercontent.com/assets/13518570/9020921/ac0288b8-37dd-11e5-8d5a-b472c8fe7470.PNG)

**Comment by bogdi1988** [comment]
OK! Got the pointer stabilized! Seems like the cameras are really sensitive to distance and that I need at least 6 inches from camera to keyboard. Also, took the track_plus.exe app and disabled scaling on high DPI settings. I now have pointer going on the entire screen!
Clicks don't work, but it seems to register when I 'click' as the pointer turns green.
![capture2](https://cloud.githubusercontent.com/assets/13518570/9020941/d404fed0-37de-11e5-918a-d7d83af87781.PNG)
![capture 3](https://cloud.githubusercontent.com/assets/13518570/9020946/0a169150-37df-11e5-931e-44b5678207ce.PNG)

**Comment by sheley1998** [comment]
Yeah clicks have been disabled during debug. I'll try to borrow a high DPI display so that I can see what the problem is on 4K displays. Can you open up a new issue on GitHub regarding this issue? Thanks so much!

**Comment by bogdi1988** [comment]
@touchhope try to move the camera back a bit and the app will pass the wiggle screen easier. also, disable the high DPI support!
@sheley1998 i will open an new bug. I also opened another one regarding multi monitor support

**Comment by bogdi1988** [comment]
@sheley1998 issue opened. Please take a look at it. I found another issue with the pointer not being the highest 'layer' in the display. Separate issue opened for it.

**Comment by sheley1998** [comment]
Yes the debug pointer executable hasn't been code-signed. Pointer can only be displayed above all UI elements after the executable is signed with a special permit purchased from Microsoft. I'll post a download link for the code signed pointer here in a bit.

**Comment by bogdi1988** [comment]
Hmm... weird. I have had it sometimes over the UI elements. Let me try and get a screenshot.

**Comment by sheley1998** [comment]
I think it works sometimes over regular UI elements. The code-signed executable always works, and even stays above microsofts full screen Modern apps in Win8

**Comment by bogdi1988** [comment]
Gotcha. Ok :) Looking forward for next item to test.

**Comment by sheley1998** [comment]
Microsoft created a level of permission called UIAccess for WPF applications. This permission is only granted to "safe apps", which basically means we've had to pay them.

**Comment by alexanderzeillinger** [comment]
Have you posted a link to the code-signed executable yet?
